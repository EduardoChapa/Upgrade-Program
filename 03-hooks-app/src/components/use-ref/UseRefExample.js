import React, { useState } from 'react';
import MultipleCustomHooks from '../use-fetch/MultipleCustomHooks';

function UseRefExample() {

    const [show, setShow] = useState(false);

    return(
        <div>
            <h1>useRef Example</h1>
            <hr />
            { show && <MultipleCustomHooks /> }
            <br/>
            <button
                className='btn btn-primary mt-5'
                onClick={ () => { setShow(!show) } }>
                Show/Hide
            </button>
        </div>
    );
}

export default UseRefExample;