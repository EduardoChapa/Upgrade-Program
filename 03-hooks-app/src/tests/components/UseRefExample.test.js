import React from 'react';
import { shallow } from 'enzyme';
import UseRefExample from '../../components/use-ref/UseRefExample';

describe('Testing UseRefExample', () => {

    const wrapper = shallow(<UseRefExample />);

    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    });
    
    test('should display correctly', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
});

