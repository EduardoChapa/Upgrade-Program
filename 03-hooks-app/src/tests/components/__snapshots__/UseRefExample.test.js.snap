// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UseRefExample should display correctly 1`] = `
<div>
  <h1>
    useRef Example
  </h1>
  <hr />
  <br />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
